### DISCUSSION TITLE
Mastodon OAuth access token could not be obtained: Error getting OAuth access token Client error: `POST https://%40soulfire@example.com/oauth/token` resulted in a `401 Unauthorized` response:

### DISCUSSION THREAD

**Discussion by S0ulf3re** [mastodon/mastodon#18477] [Q&A]
I’m trying to use the mastodon integration feature in Nextcloud that lets you view your mastodon notifications and timeline from the dashboard. But every time I hit “Authorize” from mastodon, I get an error. I’ve already opened an issue [at the Nextcloud extension](https://github.com/nextcloud/integration_mastodon/issues/27), but I just want to make sure I’m not doing something wrong. Like a part of me is saying the Nextcloud team forgot to document a step in the process. This is the error I get from Nextcloud:

Mastodon OAuth access token could not be obtained: Error getting OAuth access token Client error: `POST https://%40soulfire@toony.club/oauth/token` resulted in a `401 Unauthorized` response:
{"error":"invalid_client","error_description":"Client authentication failed due to unknown client, no client authenticat (truncated...)

Do I just need to create something in the development settings of my mastodon profile?

**Comment by trwnh** [comment]
You seem to be putting your username@domain in there instead of just your domain. Either that, or Nextcloud is doing something wrong...
